api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
+    return jsonify(member_schema.dump(member))
 
 
-@groups.route('/members/<member_id>', methods=['PATCH'])
+@groups.route('/groups/<int:group_id>/members/<int:person_id>', methods=['PATCH'])
 @jwt_required
-def update_member(member_id):
+def update_member(group_id, person_id):
+    member_schema = MemberSchema(exclude=['group_id', 'person_id'])
     try:
-        valid_member = member_schema.load(request.json)
+        valid_attributes = member_schema.load(request.json, partial=True)
     except ValidationError as err:
         return jsonify(err.messages), 422
 
-    member = db.session.query(Member).filter_by(id=member_id).first()
-
+    member = db.session.query(Member).filter_by(group_id=group_id, person_id=person_id).first()
     if member is None:
-        return jsonify(msg="No members found"), 404
+        return jsonify(f"Member with group_id #{group_id} and person_id #{person_id} does not exist"), 404
 
-    for key, val in valid_member.items():
+    for key, val in valid_attributes.items():
         setattr(member, key, val)
 
+    db.session.add(member)
     db.session.commit()
-    return jsonify(member_schema.dump(member))
 
+    return jsonify(member_schema.dump(member)), 200
 
-@groups.route('/members/activate/<member_id>', methods=['PUT'])
-@jwt_required
-def activate_member(member_id):
-    member = db.session.query(Member).filter_by(id=member_id).first()
 
-    if member is None:
-        return jsonify(msg="Member not found"), 404
-
-    setattr(member, 'active', True)
-    db.session.commit()
-    return jsonify(member_schema.dump(member))
-
-
-@groups.route('/members/deactivate/<member_id>', methods=['PUT'])
+@groups.route('/groups/<int:group_id>/members/<int:person_id>', methods=['DELETE'])
 @jwt_required
-def deactivate_member(member_id):
-    member = db.session.query(Member).filter_by(id=member_id).first()
+def delete_member(group_id, person_id):
+    member = db.session.query(Member).filter_by(group_id=group_id, person_id=person_id).first()
 
     if member is None:
-        return jsonify(msg="Member not found"), 404
+        return jsonify(f"Member with group_id #{group_id} and person_id #{person_id} does not exist"), 404
 
-    setattr(member, 'active', False)
+    db.session.delete(member)
     db.session.commit()
-    return jsonify(member_schema.dump(member))
 
+    return "Deleted successfully", 204
 
 # ---- Attendance
 
 attendance_schema = AttendanceSchema()
 
-
-@groups.route('/attendance', methods=['POST'])
+@groups.route('/meetings/<int:meeting_id>/attendances/<int:person_id>', methods=['POST', 'PUT', 'PATCH'])
 @jwt_required
-def create_attendance():
-    try:
-        valid_attendance = attendance_schema.load(request.json)
-    except ValidationError as err:
-        return jsonify(err.messages), 422
+def create_attendance(meeting_id, person_id):
+    if not db.session.query(Person).filter_by(id=person_id).first():
+        return jsonify(f"Person with person_id #{person_id} does not exist"), 404
```

Comment:
Thinking about these error messages, we don't want to report internal database keys to the end user, but we _do_ probably want to log these kinds of errors someplace. I'd bet that there are several standard packages for logging errors and warnings to a file that could be inspected by the system administrator. I've created an issue about this.